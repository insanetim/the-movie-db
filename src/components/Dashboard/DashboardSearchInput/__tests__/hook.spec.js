import React from 'react'
import { useNavigate } from 'react-router-dom'
import { act, renderHook } from '@testing-library/react'

import useContainer from '../hook'

jest.mock('src/store/dashboard/selectors', () => ({
  searchQuerySelector: jest.fn(() => null)
}))

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useNavigate: jest.fn()
}))
const navigate = jest.fn()
useNavigate.mockReturnValue(navigate)

describe('DashboardSearchInput useContainer hook', () => {
  let result = null
  const props = 'test/searchQuery'
  const setState = jest.fn()
  const useStateSpy = jest.spyOn(React, 'useState')
  useStateSpy.mockImplementation(initialState => [initialState, setState])

  beforeEach(() => {
    ;({ result } = renderHook(() => useContainer(props)))

    jest.clearAllMocks()
  })

  it('matches snapshot', () => {
    expect(result.current).toMatchSnapshot()
  })

  it('checks `handleChange` method', () => {
    act(() => {
      result.current.handleChange({
        target: {
          value: 'test/search'
        }
      })
    })

    expect(setState).toHaveBeenCalledWith('test/search')
  })

  it('checks `handleSearch` method with value', () => {
    act(() => {
      result.current.handleSearch('value')
    })

    expect(navigate).toHaveBeenCalledWith({ pathname: '/', search: 'search=value' })
  })

  it('checks `handleSearch` method without value', () => {
    act(() => {
      result.current.handleSearch('')
    })

    expect(navigate).toHaveBeenCalledWith('/')
  })
})
