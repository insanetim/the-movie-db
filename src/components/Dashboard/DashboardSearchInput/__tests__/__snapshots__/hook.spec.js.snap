// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DashboardSearchInput useContainer hook matches snapshot 1`] = `
{
  "currentValue": "test/searchQuery",
  "handleChange": [Function],
  "handleSearch": [Function],
}
`;
